pkg/project/sync.go

```diff
@@ -457,3 +429,79 @@ func handleMissingProjectDir(httpClient utils.HTTPClient, connection *connection
 
 	return nil
 }
+
+func findNewFiles(client utils.HTTPClient, projectID string, beforefiles []string, afterfiles []string, projectPath string, connection *connections.Connection, conURL string) []string {
+	var newfiles []string
+	for _, filename := range afterfiles {
+		if !existsIn(filename, beforefiles) {
+			fullPath := filepath.Join(projectPath, filename)
+			syncFile(&http.Client{}, projectID, projectPath, fullPath, connection, conURL)
+			newfiles = append(newfiles, filename)
+		}
+	}
+	return newfiles
+}
+
+func existsIn(value string, slice []string) bool {
+	for _, item := range slice {
+		if item == value {
+			return true
+		}
+	}
+	return false
+}
+
+func syncFile(client utils.HTTPClient, projectID string, projectPath string, path string, connection *connections.Connection, conURL string) UploadedFile {
+	// use ToSlash to try and get both Windows and *NIX paths to be *NIX for pfe
+	relativePath := filepath.ToSlash(path[(len(projectPath) + 1):])
+	uploadResponse := UploadedFile{
+		FilePath:   relativePath,
+		Status:     "Failed",
+		StatusCode: 0,
+	}
+	// Retrieve file info
+	fileStat, err := os.Stat(path)
+	if err != nil {
+		return uploadResponse
+	}
+
+	fileContent, err := ioutil.ReadFile(path)
+	// Return here if there is an error reading the file
+	if err != nil {
+		return uploadResponse
+	}
+
+	fileUploadBody := FileUploadMsg{
+		IsDirectory:  fileStat.IsDir(),
+		Mode:         uint(fileStat.Mode().Perm()),
+		RelativePath: relativePath,
+		Message:      "",
+	}
+
+	var buffer bytes.Buffer
+	zWriter := zlib.NewWriter(&buffer)
+	zWriter.Write([]byte(fileContent))
+
+	zWriter.Close()
+	encoded := base64.StdEncoding.EncodeToString(buffer.Bytes())
+	fileUploadBody.Message = encoded
+
+	buf := new(bytes.Buffer)
+	json.NewEncoder(buf).Encode(fileUploadBody)
+
+	projectUploadURL := conURL + "/api/v1/projects/" + projectID + "/upload"
+	// TODO - How do we handle partial success?
+	request, err := http.NewRequest("PUT", projectUploadURL, bytes.NewReader(buf.Bytes()))
```

Comment:
I don't think it's likely to happen and this isn't new code but we don't check `err` here.

Reply:
I tried not to change anything if I didn't need to, can be addressed later